src/go/rpk/pkg/cli/generate/app.go

```diff
@@ -0,0 +1,430 @@
+	ScramBits  string
+	SeedServer []string
+	Username   string
+	IsTLS      bool
+}
+
+func newAppCmd(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		langFlag string
+		outPath  string
+		saslCred string
+		noUser   bool
+	)
+	cmd := &cobra.Command{
+		Use:   "app",
+		Short: "Generate a sample application to connect with Redpanda",
+		Long:  appHelpText,
+		Run: func(cmd *cobra.Command, args []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			isInteractive := langFlag == "" // The rest of the flags are optional.
+			if !isInteractive {
+				if saslCred != "" {
+					out.Die("--new-sasl-credentials flag can not be used without --lang flag")
+				} else if noUser {
+					out.Die("--no-user flag can not be used without --lang flag")
+				}
+			}
+
+			var lang, user, pass string
+			if isInteractive {
+				lang, user, pass, err = runInteractive(cmd.Context(), fs, p)
+				out.MaybeDieErr(err)
+			} else {
+				lang, user, pass, err = runWithFlags(cmd.Context(), fs, p, langFlag, saslCred, noUser)
+				out.MaybeDieErr(err)
+			}
+			data := tmplData{
+				SeedServer: p.KafkaAPI.Brokers,
+				Username:   user,
+				Password:   pass,
+				IsTLS:      p.KafkaAPI.TLS != nil,
+				ScramBits:  scramBits(p.KafkaAPI.SASL),
+			}
+
+			f, ok := fileTplMap[lang]
+			// Unlikely since we check that the user only selects the allowed
+			// languages, but better be safe.
+			if !ok {
+				out.Die("unable to create a sample app for the selected language: %q", lang)
+			}
+			dir, err := extractTarGzAndReplace(tplFS, fs, data, filepath.Join(TemplateDir, f.Location), outPath)
+			out.MaybeDie(err, "unable to extract tarball: %v", err)
+
+			if user != "" {
+				fmt.Printf("Successfully created the sample app with user %q in:\n\n", user)
+			} else {
+				fmt.Printf("Successfully created the sample app in:\n\n")
+			}
+			err = printInstructions(fs, dir, lang)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "unable to print the app instructions: %v; you may find the instructions in the README file inside the generated app\n", err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&langFlag, "lang", "", "The language you want the code sample to be generated with")
+	cmd.Flags().StringVarP(&outPath, "output", "o", "", "The path where the app will be written")
+	cmd.Flags().StringVar(&saslCred, "new-sasl-credentials", "", "If provided, rpk will generate and use this credentials (<user>:<password>)")
+	cmd.Flags().BoolVar(&noUser, "no-user", false, "Generates the sample app without SASL user")
+
+	return cmd
+}
+
+func runInteractive(ctx context.Context, fs afero.Fs, p *config.RpkProfile) (lang, user, pass string, err error) {
+	lang, err = out.Pick(languages, "Choose a language for your sample app:")
+	if err != nil {
+		return "", "", "", fmt.Errorf("error picking the language: %v", err)
+	}
+	lang = strings.ToLower(lang)
+
+	optMap := map[string]int{
+		"Create new user with read-write ACLs": optCreate,
```

Comment:
The acls are a bit more generous than read/write, wdyt about `Create a new user with admin ACLs`?